stream-chat-android/src/main/java/com/getstream/sdk/chat/adapter/MessageListItemDiffCallback.kt

```diff
@@ -1,13 +1,59 @@
 package com.getstream.sdk.chat.adapter
 
 import androidx.recyclerview.widget.DiffUtil
+import io.getstream.chat.android.client.models.User
 
 internal object MessageListItemDiffCallback : DiffUtil.ItemCallback<MessageListItem>() {
     override fun areItemsTheSame(oldItem: MessageListItem, newItem: MessageListItem): Boolean {
         return oldItem.getStableId() == newItem.getStableId()
     }
 
     override fun areContentsTheSame(oldItem: MessageListItem, newItem: MessageListItem): Boolean {
-        return oldItem == newItem
+        var equal = true
+        when (oldItem) {
+            is MessageListItem.MessageItem -> {
+                newItem as MessageListItem.MessageItem
+                equal = if (oldItem.message.text != newItem.message.text) {
+                    false
+                } else if (oldItem.message.reactionScores != newItem.message.reactionScores) {
+                    false
+                } else if (oldItem.message.reactionCounts != newItem.message.reactionCounts) {
+                    false
+                } else if (oldItem.message.attachments != newItem.message.attachments) {
+                    false
+                } else if (oldItem.message.replyCount != newItem.message.replyCount) {
+                    false
+                } else if (oldItem.message.syncStatus != newItem.message.syncStatus) {
+                    false
+                } else if (oldItem.message.deletedAt != newItem.message.deletedAt) {
+                    false
+                } else if (oldItem.positions != newItem.positions) {
+                    false
+                } else oldItem.messageReadBy.map { it.getUserId() } == newItem.messageReadBy.map { it.getUserId() }
+            }
+            is MessageListItem.DateSeparatorItem, is MessageListItem.ThreadSeparatorItem, is MessageListItem.LoadingMoreIndicatorItem -> equal = true
+            is MessageListItem.TypingItem -> oldItem.users.map(User::id) == ((newItem) as MessageListItem.TypingItem).users.map(User::id)
+            is MessageListItem.ReadStateItem -> oldItem.reads.map { it.getUserId() } == ((newItem) as MessageListItem.ReadStateItem).reads.map { it.getUserId() }
```

Comment:
Shouldn't we update `equal` variable here?  Maybe we can also get rid of `equal` variable and return the boolean directly from `when` expression?

Reply:
Good catch!!!